File: primefactors/kotlin/src/main/kotlin/primefactors/PrimeFactors.kt
package primefactors

fun primeFactors(number: Int): List<Int> {
    val factors = mutableListOf<Int>()
    var n = number

    for (d in 2..number) {
        while (n.rem(d) == 0) {
            factors.add(d).also { n /= d }
        }

        if (n == 1) return factors
    }

    return emptyList()
}

File: primefactors/javascript/js/primefactors.js
"use strict";

export function factors(number) {
    const factors = [];
    let n = number;

    for (let i = 2; i <= number; i++) {
        while (n % i === 0) {
            factors.push(i);
            n /= i;
        }

        if (n === 1) return factors;
    }

    return factors;
}